Mathlib/MeasureTheory/Measure/Haar/Quotient.lean

```diff
@@ -147,12 +146,167 @@ theorem MeasureTheory.IsFundamentalDomain.map_restrict_quotient [Subgroup.Normal
+  · exact fun i ↦ (hf₁.1.comp_quasiMeasurePreserving
+      (measurePreserving_smul i μ).quasiMeasurePreserving).restrict
+  · rw [← h𝓕.lintegral_eq_tsum'' (fun x ↦ ‖f x‖₊)]
+    exact ne_of_lt hf₁.2
+
+/-- This is the **Unfolding Trick**: Given a subgroup `Γ` of a group `G`, the integral of a
+  function `f` on `G` times the lift to `G` of a function `g` on the quotient `G ⧸ Γ` with respect
+  to a right-invariant measure `μ` on `G`, is equal to the integral over the quotient of the
+  automorphization of `f` times `g`. -/
+lemma QuotientGroup.integral_mul_eq_integral_automorphize_mul {K : Type _} [NormedField K]
+    [NormedSpace ℝ K] [μ.IsMulRightInvariant] {f : G → K}
+    (f_ℒ_1 : Integrable f μ) {g : G ⧸ Γ → K} (hg : AEStronglyMeasurable g μ_𝓕)
+    (g_ℒ_infinity : essSup (fun x ↦ ↑‖g x‖₊) μ_𝓕 ≠ ∞)
+    (F_ae_measurable : AEStronglyMeasurable (QuotientGroup.automorphize f) μ_𝓕) :
+    ∫ x : G, g (x : G ⧸ Γ) * (f x) ∂μ
+      = ∫ x : G ⧸ Γ, g x * (QuotientGroup.automorphize f x) ∂μ_𝓕 := by
+  let π : G → G ⧸ Γ := QuotientGroup.mk
+  have meas_π : Measurable π := continuous_quotient_mk'.measurable
+  have H₀ : QuotientGroup.automorphize ((g ∘ π) * f) = g * (QuotientGroup.automorphize f) := by
+    exact QuotientGroup.automorphize_smul_left f g
+  calc ∫ (x : G), g (π x) * (f x) ∂μ =
+        ∫ (x : G ⧸ Γ), QuotientGroup.automorphize ((g ∘ π) * f) x ∂μ_𝓕 := ?_
+    _ = ∫ (x : G ⧸ Γ), g x * (QuotientGroup.automorphize f x) ∂μ_𝓕 := by simp [H₀]
+  have H₁ : Integrable ((g ∘ π) * f) μ
+  · have : AEStronglyMeasurable (fun (x : G) ↦ g (x : (G ⧸ Γ))) μ
+    · refine (AEStronglyMeasurable.mono' hg ?_).comp_measurable meas_π
+      exact h𝓕.absolutelyContinuous_map
+    refine Integrable.essSup_smul f_ℒ_1 this ?_
+    have hg' : AEStronglyMeasurable (fun x ↦ (‖g x‖₊ : ℝ≥0∞)) μ_𝓕 :=
+      (ENNReal.continuous_coe.comp continuous_nnnorm).comp_aestronglyMeasurable hg
+    rw [← essSup_comp_quotientGroup_mk h𝓕 hg'.aemeasurable]
+    exact g_ℒ_infinity
+  have H₂ : AEStronglyMeasurable (QuotientGroup.automorphize ((g ∘ π) * f)) μ_𝓕
+  · simp_rw [H₀]
+    exact hg.mul F_ae_measurable
+  apply QuotientGroup.integral_eq_integral_automorphize h𝓕 H₁ H₂
+
+end
+
+section
+
+variable {G' : Type _} [AddGroup G'] [MeasurableSpace G'] [TopologicalSpace G']
+  [TopologicalAddGroup G'] [BorelSpace G']
+  {μ' : Measure G'}
+  {Γ' : AddSubgroup G'}
+  [Countable Γ'] [MeasurableSpace (G' ⧸ Γ')] [BorelSpace (G' ⧸ Γ')]
+  {𝓕' : Set G'}
+
+local notation "μ_𝓕" => Measure.map (@QuotientAddGroup.mk G' _ Γ') (μ'.restrict 𝓕')
+
+/-- This is the **Unfolding Trick**: Given an additive subgroup `Γ'` of an additive group `G'`, the
+  integral of a function `f` on `G'` times the lift to `G'` of a function `g` on the quotient
+  `G' ⧸ Γ'` with respect to a right-invariant measure `μ` on `G'`, is equal to the integral over
+  the quotient of the automorphization of `f` times `g`. -/
+lemma QuotientAddGroup.integral_mul_eq_integral_automorphize_mul {K : Type _} [NormedField K]
```

Comment:
what happens with `@[to_additive]` automatically here? should this be a bug report?

Reply:
I think there are just too many things going on at once for `@[to_additive]` to work out which things should be additivized and which should remain the same...